bigchaindb_driver/offchain.py

```diff
@@ -0,0 +1,229 @@
+"""
+Module for operations that can be performed "offchain", meaning without
+a connection to one or more  BigchainDB federation nodes.
+
+"""
+import logging
+from functools import singledispatch
+
+from bigchaindb.common.transaction import (
+    Fulfillment as BdbFulfillment,
+    Transaction,
+    TransactionLink,
+)
+from bigchaindb.common.exceptions import KeypairMismatchException
+from cryptoconditions import Fulfillment
+
+from .exceptions import BigchaindbException, MissingSigningKeyError
+from .utils import (
+    CreateOperation,
+    TransferOperation,
+    _normalize_asset,
+    _normalize_operation,
+)
+
+logger = logging.getLogger(__name__)
+
+
+@singledispatch
+def _prepare_transaction(operation,
+                         owners_before=None,
+                         owners_after=None,
+                         asset=None,
+                         metadata=None,
+                         inputs=None):
+    raise BigchaindbException((
+        'Unsupported operation: {}. '
+        'Only "CREATE" and "TRANSFER" are supported.'.format(operation)))
+
+
+@_prepare_transaction.register(CreateOperation)
+def _prepare_create_transaction_dispatcher(operation, **kwargs):
+    del kwargs['inputs']
+    return prepare_create_transaction(**kwargs)
+
+
+@_prepare_transaction.register(TransferOperation)
+def _prepare_transfer_transaction_dispatcher(operation, **kwargs):
+    del kwargs['owners_before']
```

Comment:
Yes, the registered function will receive all the arguments of the generic function (the one decorated with `@singledispatch`). The arguments that are not needed are simply deleted.